Minor fixes and updates to CLI

ersonp opened the pull request:
Did you run `make format && make check`?
yes

Fixes #	

 Changes:	
- Changed Global variables to Local
- Minor updates in cmd flags
- Reverted struct `Common` to use the oath variable from it's own struct
- Removed unused param `args` from funcs
- Updated mac and windows path variables and PackageConfig
- Deleted unused code

How to test this PR:

ersonp committed:
Fix config gen

This commit fixes config gen. Instead of saving to and using a global variable skyenv.ConfigName we use the local variable output to save and update the path of the config as needed.

Also the global var skyenv.DMSGHTTPPath that was used to  save and update and use is chanegd by dmsgHTTPPath.

The code to get sk and hv from old config was moved under one block.

Also there was a but where we generated new pk-sk pair even if the config was read without any errors was fixed.

ersonp committed:
Fix skyenv values

This commit changes ConfigName and DMSGHTTPPath from var to const.

ersonp committed:
Fix visor config common

This commit fixes the struct common in the package visorconfig.

We reuse the field path in the common struct to flush the config instead of using it from the params of the method flush which was also removed.

We are going back to the way it was done before as it is much more cleaner to do it that way and we avoid introducing any variables.

ersonp committed:
Fix config gen

ersonp committed:
Update config update

This commit deletes the consts serviceconfaddr and testconfaddr and uses utilenv.ServiceConfAddr and utilenv.TestServiceConfAddr instead.

ersonp committed:
Fix skywire-visor root command

This commit uses local vars skywire, workDir and root instead of global vars skyenv.Skywire, skyenv.Wd, skyenv.Root since they we used within the same package. We also delete them from skyenv.

We also use local var confPath instead of global var skyenv.ConfigName.

ersonp committed:
Fix func MakeBaseConfig

This commit adds the param dmsgHTTPServersList to the func MakeBaseConfig instead of using it as a local varialbe.

ersonp committed:
Make check

ersonp commented on cmd/skywire-cli/commands/config/gen.go:
```diff
@@ -121,24 +123,25 @@ var genConfigCmd = &cobra.Command{
 		}
 		if dmsgHTTP {
+			dmsgHTTPPath := skyenv.DMSGHTTPPath
 			if pkgEnv {
```
@mrpalide @0pcom do we need to use a different dmsghttp config path for package?

0pcom replied:
`/opt/skywire/dmsghttp-config.json` for the linux packages
something different for windows & mac I'm sure

mrpalide replied:
Main problem here is when trying to run `skywire-cli config gen ...` in a path different from location of files.
As Moses said for linux packages is `/opt/skywire/dmsghttp-config.json`.
For windows be good if we disable choosing install path by users, and just use `C:/Program Files/Skywire/dmsghttp-config.json`. Also same for Mac but I don't know about it.

ersonp replied:
I can add the windows and mac paths. Can someone just confirm the Mac path please? And this path `C:/Program Files/Skywire` would be the same for `skywire-config.json` too correct?

mrpalide replied:
Based on Mac Installer scripts, Mac path is correct.
And about the second question, we use another output for config generated, `"%HOMEPATH%\skywire-config.json"`.

ersonp replied:
So the `skywire-config.json` and `dmsghttp-config.json` are in a different location in the windows installer?

mrpalide replied:
Yes.

ersonp commented on pkg/visor/visorconfig/config.go:
```diff
@@ -111,13 +110,20 @@ func MakeBaseConfig(common *Common, testEnv bool, dmsgHTTP bool, services *Servi
 	if dmsgHTTP {
-		serversListJSON, err := ioutil.ReadFile(skyenv.DMSGHTTPPath)
+		dmsgHTTPPath := skyenv.DMSGHTTPPath
+		if pkgEnv {
```
> @mrpalide @0pcom do we need to use a different dmsghttp config path for package?

same question here

0pcom commented on cmd/skywire-cli/commands/config/gen.go:
```diff
@@ -67,7 +68,7 @@ func init() {
 	genConfigCmd.Flags().StringVarP(&selectedOS, "os", "k", skyenv.OS, "(linux / macos / windows) paths")
 	genConfigCmd.Flags().BoolVarP(&stdout, "stdout", "n", false, "write config to stdout")
-	genConfigCmd.Flags().StringVarP(&output, "out", "o", "", "output config default:"+skyenv.ConfigName)
+	genConfigCmd.Flags().StringVarP(&output, "out", "o", skyenv.ConfigName, "output config default:"+skyenv.ConfigName)
```
this change messes up the formatting of the help menu. The way it was before was intentional for that reason

![image](https://user-images.githubusercontent.com/36607567/162284438-0de0db5d-fe0d-4222-bdeb-51176ab15df1.png)

previously:

![image](https://user-images.githubusercontent.com/36607567/162284941-924ff40a-dcd4-4ecb-a996-b67e2d10761e.png)

ersonp replied:
Fixed
![image](https://user-images.githubusercontent.com/38801159/162367666-24472ddd-4fa9-405f-aff3-1c6cd1ce96f7.png)

0pcom commented:
![image](https://user-images.githubusercontent.com/36607567/162287822-cf008758-34e2-46c9-9de2-b32d794e21b2.png)

wrong version

0pcom commented:
![image](https://user-images.githubusercontent.com/36607567/162288190-38208fff-edb6-4f4c-bb53-01ed94cbb2cd.png)
same text wrap-around issue for the visor help menu as with config gen on the cli

ersonp committed:
Fix formatting on help menu

ersonp commented:
> ![image](https://user-images.githubusercontent.com/36607567/162287822-cf008758-34e2-46c9-9de2-b32d794e21b2.png)
> 
> wrong version

I haven't changed anything about the version. You might need to pull the latest tags.
![image](https://user-images.githubusercontent.com/38801159/162367506-d8670bb7-fd75-4b28-a507-c81544c7d58e.png)

ersonp commented:
> ![image](https://user-images.githubusercontent.com/36607567/162288190-38208fff-edb6-4f4c-bb53-01ed94cbb2cd.png) same text wrap-around issue for the visor help menu as with config gen on the cli

Fixed
![image](https://user-images.githubusercontent.com/38801159/162367808-395d677f-ecdc-4934-9f45-c31384771972.png)

ersonp committed:
Fix systray

ersonp committed:
Fix DmsghttpPath and DMSGHTTPPath

ersonp committed:
Fix values

ersonp committed:
Fix skyenv.BuildInfo

This commit deletes the global var skyenv.BuildInfo and instead creates a local var in skywirevisor root.

ersonp committed:
Requested changes to windows installer

ersonp committed:
Fix dmsgpty

This commit uses dmsgpty.DefaultCLIAddr() instead of CLIAddr() and deletes the dmsgpty.go files as they are duplicates of the dmsgpty from the package dmsg.

mrpalide reviewed (approved):
Approved for Linux and Windows.

0pcom commented:
approved for macos
